rmw_zenoh_cpp/src/detail/graph_cache.cpp

```diff
@@ -423,31 +595,30 @@ rmw_ret_t GraphCache::get_node_names(
       std::move(free_enclaves_lambda));
   }
 
-  // TODO(Yadunund): Remove this printout.
-  const std::string & graph_str = YAML::Dump(graph_);
-  RCUTILS_LOG_WARN_NAMED("rmw_zenoh_cpp", "[graph]\n%s\n", graph_str.c_str());
   // Fill node names, namespaces and enclaves.
   std::size_t j = 0;
-  for (auto it = graph_.begin(); it != graph_.end(); ++it) {
-    const auto & node_name = it->first.as<std::string>();
-    const auto & yaml_node = it->second;
-    node_names->data[j] = rcutils_strdup(node_name.c_str(), *allocator);
-    if (!node_names->data[j]) {
-      return RMW_RET_BAD_ALLOC;
-    }
-    node_namespaces->data[j] = rcutils_strdup(
-      yaml_node["namespace"].as<std::string>().c_str(), *allocator);
-    if (!node_namespaces->data[j]) {
-      return RMW_RET_BAD_ALLOC;
-    }
-    if (enclaves) {
-      enclaves->data[j] = rcutils_strdup(
-        yaml_node["enclaves"].as<std::string>().c_str(), *allocator);
-      if (!enclaves->data[j]) {
+  for (auto ns_it = graph_.begin(); ns_it != graph_.end(); ++ns_it) {
+    const std::string & ns = ns_it->first;
+    for (auto node_it = ns_it->second.begin(); node_it != ns_it->second.end(); ++node_it) {
+      const auto node = node_it->second;
+      node_names->data[j] = rcutils_strdup(node->name.c_str(), *allocator);
+      if (!node_names->data[j]) {
         return RMW_RET_BAD_ALLOC;
       }
+      node_namespaces->data[j] = rcutils_strdup(
+        ns.c_str(), *allocator);
+      if (!node_namespaces->data[j]) {
+        return RMW_RET_BAD_ALLOC;
+      }
+      if (enclaves) {
+        enclaves->data[j] = rcutils_strdup(
+          node->enclave.c_str(), *allocator);
+        if (!enclaves->data[j]) {
+          return RMW_RET_BAD_ALLOC;
+        }
+      }
+      ++j;
```

Comment:
General comment for this method. 
I wasn't able to retrieve the list of nodes via `ros2 node list`. I verified and we de call this method from `rmw_get_node_names`. I couldn't debug yet to see what is going on.

Reply:
Surprised to hear this. Was the router running? Without the router the node that gets spun via `ros2 node list` will not get gossip information about other nodes.

Reply:
This ended up working for me, as long as the router is running.  @francocipollone give it another shot with the latest, see if you can make it work.

Reply:
Confirmed --> https://github.com/ros2/rmw_zenoh/pull/66#issuecomment-1817112633